shell/promptRemove/mixin/roleDeletionCheck.js

```diff
@@ -0,0 +1,91 @@
+import { mapState, mapGetters } from 'vuex';
+import { resourceNames } from '@shell/utils/string';
+import { MANAGEMENT } from '@shell/config/types';
+
+export default {
+  data() {
+    return {
+      warning:             '',
+      info:                '',
+    };
+  },
+
+  computed:   {
+    ...mapState('action-menu', ['toRemove']),
+    ...mapGetters({ t: 'i18n/t' }),
+
+    names() {
+      return this.toRemove.map(obj => obj.nameDisplay).slice(0, 5);
+    },
+
+    plusMore() {
+      const remaining = this.toRemove.length - this.names.length;
+
+      return this.t('promptRemove.andOthers', { count: remaining });
+    },
+  },
+  watch: {
+    value: {
+      handler(neu) {
+        this.handleRoleDeletionCheck(neu, neu[0].type);
+      },
+      immediate: true
+    }
+  },
+  methods: {
+    resourceNames,
+    async handleRoleDeletionCheck(rolesToRemove, resourceType) {
+      this.warning = '';
+      let resourceToCheck;
+      let propToMatch;
+      let numberOfRolesWithBinds = 0;
+      let numberUniqueUsersWithBinds = 0;
+
+      this.info = this.t('rbac.globalRoles.waiting', { count: rolesToRemove.length });
+
+      switch (resourceType) {
+      case MANAGEMENT.GLOBAL_ROLE:
+        resourceToCheck = MANAGEMENT.GLOBAL_ROLE_BINDING;
+        propToMatch = 'globalRoleName';
+        break;
+      default:
+        resourceToCheck = MANAGEMENT.CLUSTER_ROLE_TEMPLATE_BINDING;
+        propToMatch = 'roleTemplateName';
+        break;
+      }
+
+      try {
+        const data = await this.$store.dispatch('management/request', {
```

Comment:
This is a nitpick, but could we call this variable something other than `data`? I just have a rough time when trying to understand what `data.data` is referring to.